prometheus/timer_test.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"testing"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+func TestTimerObserve(t *testing.T) {
+	his := NewHistogram(
+		HistogramOpts{
+			Name: "test_histogram",
+		},
+	)
+
+	stop := NewTimer()
+	stop(his)
```

Comment:
The way this looks in Java is

```
static final Summary requestLatency = Summary...

Summary.Timer requestTimer = requestLatency.startTimer();
requestTimer.observeDuration();
```

We should keep to the same naming and structure for consistency.